clang/test.cpp

```
@@ -0,0 +1,88 @@
+int globalCounter = 0;
+
+#define MAX_BUFFER 256
+
+int Add(int a, int b) {
+    return a + b;
+}
+
+void leakyFunction() {
+    int* data = (int*)malloc(sizeof(int) * 10);
+    data[0] = 42;
+}
```

Cppcheck (2.20.0)

[error] Memory leak

(memleak)